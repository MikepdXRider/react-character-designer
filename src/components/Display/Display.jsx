import React from 'react'

const globalArticleStyle = {
    width: '150px',
    height: '150px',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'auto'
}

export default function Display({head, torso}) {
    return (
        <section>
            <article 
            style={{
                ...globalArticleStyle,
                backgroundImage: `url(${head}-hat.png)`
                }}>
            </article>
            <article 
            style={{
                ...globalArticleStyle,
                backgroundImage: `url(${torso}-torso.png)`
                }}>

            </article>
        </section>
    )
}
