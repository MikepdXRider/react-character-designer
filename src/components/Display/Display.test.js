import {screen, render} from '@testing-library/react'
import Display from './Display.jsx'

it('renders display component', () => {
    // render enables us to pseudo-render a component. This also lets us use the screen method later on
    const {container} = render(<Display head='head' torso='torso' bottom='bottom' catchphraseListArr={['item1', 'item2', 'item3']} />);
    //data-testid attributes hardcoded onto jsx tags act as ids that can be grabbed here.
    const articleEl = screen.getByTestId('article');
    const catchphraseEl = screen.getByTestId('item1');
    // expect the above elements to be in the rendered document.
    expect(articleEl).toBeInTheDocument();
    expect(catchphraseEl).toBeInTheDocument();
    // expect container to match the produced snapshot(first test generates, all other test compare)
    expect(container).toMatchSnapshot();
})