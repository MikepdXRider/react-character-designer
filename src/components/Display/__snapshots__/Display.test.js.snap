// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders display component 1`] = `
<div>
  <section>
    <article
      data-testid="head"
      style="width: 150px; height: 150px; background-repeat: no-repeat; background-size: auto; background-image: url(head.png);"
    />
    <article
      data-testid="torso"
      style="width: 150px; height: 150px; background-repeat: no-repeat; background-size: auto; background-image: url(torso.png);"
    />
    <article
      data-testid="bottom"
      style="width: 150px; height: 150px; background-repeat: no-repeat; background-size: auto; background-image: url(bottom.png);"
    />
  </section>
  <section>
    <p>
      item1
    </p>
    <p>
      item2
    </p>
    <p>
      item3
    </p>
  </section>
</div>
`;
